### PR TITLE
docs: clarify success/failure transform tests

### PR DESCRIPTION
**Pull request by hoffa**
### Description of changes

I didn't find any place where error transforms were explained, so here it is.

### Checklist

- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: DEVELOPMENT_GUIDE.md [hunk 1]
```diff
@@ -152,8 +154,13 @@ The script automatically updates hardcoded ARN partitions to match the output pa
 python3 bin/add_transform_test.py --template-file template.yaml --disable-update-partition
 ```
 
-Note that please always check the generated output is as expected. This tool does not guarantee correct output.
+Please always check the generated output is as expected. This tool does not guarantee correct output.
+
+#### Transform failures
+
+To test a SAM template results in a specific transform error, add the SAM template under [`tests/translator/input`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/input), and a JSON file with the expected `errorMessage` as a top-level value under [`tests/translator/output`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/output). The JSON file must be prefixed with `error_` and have the same basename as the input SAM template (e.g. `my_cool_template.yaml` for input, and `error_my_cool_template.json` for the expected error).
```
Comment (hoffa) [inline DEVELOPMENT_GUIDE.md:161]:
```suggestion
To test a SAM template results in a specific transform error, add the SAM template under [`tests/translator/input`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/input), and a JSON file with the expected `errorMessage` as a top-level value under [`tests/translator/output`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/output). The JSON file must be prefixed with `error_` and include the same basename as the input SAM template (e.g. `my_cool_template.yaml` for input, and `error_my_cool_template.json` for the expected error).
```

**Review by ssenchenko — APPROVED** [review]

File: DEVELOPMENT_GUIDE.md [hunk 2]
```diff
@@ -152,8 +154,13 @@ The script automatically updates hardcoded ARN partitions to match the output pa
 python3 bin/add_transform_test.py --template-file template.yaml --disable-update-partition
 ```
 
-Note that please always check the generated output is as expected. This tool does not guarantee correct output.
+Please always check the generated output is as expected. This tool does not guarantee correct output.
+
+#### Transform failures
+
+To test a SAM template results in a specific transform error, add the SAM template under [`tests/translator/input`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/input), and a JSON file with the expected `errorMessage` as a top-level value under [`tests/translator/output`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/output). The JSON file must be prefixed with `error_` and include the same basename as the input SAM template (e.g. `my_cool_template.yaml` for input, and `error_my_cool_template.json` for the expected error).
```
Comment (aahung) [inline DEVELOPMENT_GUIDE.md:161]: Are you sure the template file doesn't need to be `error_my_cool_template.yaml`? Example: https://github.com/aws/serverless-application-model/blob/develop/tests/translator/input/error_api_invalid_auth.yaml

Comment (hoffa) [inline DEVELOPMENT_GUIDE.md:161]:
```suggestion
To test a SAM template results in a specific transform error, add the SAM template under [`tests/translator/input`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/input), and a JSON file with the expected `errorMessage` as a top-level value under [`tests/translator/output`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/output). The JSON file must be prefixed with `error_` and have the same basename as the input SAM template (e.g. `error_my_cool_template.yaml` for input, and `error_my_cool_template.json` for the expected error).
```

Comment (hoffa) [reply DEVELOPMENT_GUIDE.md:161]: Good catch, fixed.

File: DEVELOPMENT_GUIDE.md [hunk 3]
```diff
@@ -152,8 +154,13 @@ The script automatically updates hardcoded ARN partitions to match the output pa
 python3 bin/add_transform_test.py --template-file template.yaml --disable-update-partition
 ```
 
-Note that please always check the generated output is as expected. This tool does not guarantee correct output.
+Please always check the generated output is as expected. This tool does not guarantee correct output.
+
+#### Transform failures
+
+To test a SAM template results in a specific transform error, add the SAM template under [`tests/translator/input`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/input), and a JSON file with the expected `errorMessage` as a top-level value under [`tests/translator/output`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/output). The JSON file must be prefixed with `error_` and have the same basename as the input SAM template (e.g. `error_my_cool_template.yaml` for input, and `error_my_cool_template.json` for the expected error).
```
Comment (hoffa) [inline DEVELOPMENT_GUIDE.md:161]:
```suggestion
To test a SAM template results in a specific transform error, add the SAM template under [`tests/translator/input`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/input), and a JSON file with the expected `errorMessage` as a top-level value under [`tests/translator/output`](https://github.com/aws/serverless-application-model/tree/develop/tests/translator/output). Both files must have the same basename and be prefixed with `error_` (e.g. `error_my_cool_template.yaml` for input, and `error_my_cool_template.json` for the expected error).
```

**Pull request merged**